enhancements/per-node-configuration.md

```diff
@@ -0,0 +1,850 @@
+spec:
+  nodeName: worker-2
+status:
+  matchedResources:
+    - kind: Underlay
+      name: underlay-rack-2
+  conditions:
+    - type: NodeMatched
+      status: "True"
+      reason: ConfigurationsMatched
+      message: "1 configuration resource matches this node"
+      lastTransitionTime: "2025-01-15T10:30:00Z"
+    - type: Ready
+      status: "False"
+      reason: ConfigurationFailed
+      message: "1 resource(s) failed to apply"
+      lastTransitionTime: "2025-01-15T10:30:00Z"
+    - type: Degraded
+      status: "True"
+      reason: ConfigurationFailed
+      message: "Underlay underlay-rack-2 failed: Interface eth2 not present on node"
+      lastTransitionTime: "2025-01-15T10:30:00Z"
+  failedResources:
+    - kind: Underlay
+      name: underlay-rack-2
+      message: "Interface eth2 not present on node"
+```
+
+**Note**: The Underlay, L3VNI, L2VNI, and L3Passthrough CRDs themselves do not have status fields added by this enhancement. All status reporting is centralized in the per-node RouterNodeConfigurationStatus CRD.
+
+
+### Backward Compatibility
+
+**Default Behavior**: When `nodeSelector` is `nil` or not specified, the CRDs instances applies to all nodes in the cluster, maintaining backward compatibility with existing configurations.
+
+**Migration Path**:
+N/A
+
+### Validation Rules
+
+The validation webhook enforces:
+
+1. **No Overlapping Selectors for Underlay**: Multiple Underlays cannot match the same node
+2. **Selector Validity**: Node selector must be a valid `metav1.LabelSelector` for all CRD types
+3. **Multiple instances allowed**: L3VNI, L2VNI, and L3Passthrough can have multiple instances matching the same node (no overlap validation)
+
+### RBAC Requirements
+
+The controllers require permissions to:
+
+```yaml
+- apiGroups: [""]
+  resources: ["nodes"]
+  verbs: ["get", "list", "watch"]
+- apiGroups: ["openpe.openperouter.github.io"]
+  resources: ["underlays", "l3vnis", "l2vnis", "l3passthroughs"]
+  verbs: ["get", "list", "watch"]
+- apiGroups: ["openpe.openperouter.github.io"]
+  resources: ["routernodeconfigurationstatuses"]
+  verbs: ["get", "list", "watch", "create", "update", "patch"]
+- apiGroups: ["openpe.openperouter.github.io"]
+  resources: ["routernodeconfigurationstatuses/status"]
+  verbs: ["get", "update", "patch"]
+```
+
+### Scalability Considerations
+
+- **Node Watch Overhead**: The controllers watch Node resources for label changes. In clusters with frequent node label updates, this may increase reconciliation frequency. We can improve this by making each daemonset pod reconcile only on the running node using controller-runtime cache filter mechanism.
+- **Selector Evaluation**: For each reconciliation, the controller evaluates node selectors against the node where the daemonset is running, so reconciliation is done in parallel between the nodes.
+- **Status Updates**: Status updates are per-resource, avoiding conflicts from concurrent updates
+
+## Alternatives
```

Comment:
I don't think this section is useful as it's quite a bad alternative :P

Reply:
Really bad :-), I will remove alternatives, since I forced that in.